src/os/ui/filter/op/op.js

```diff
@@ -205,7 +212,8 @@ os.ui.filter.op.Op.prototype.getLiteral = function(el) {
  */
 os.ui.filter.op.Op.prototype.matches = function(el) {
   if (el && el.length) {
-    return el[0].localName == this.localName;
+    var hint = el[0].getAttribute('hint');
+    return el[0].localName == this.localName && hint == this.matchHint;
```

Comment:
I think we always need to require both. `this.matchHint` will always default to null, so any op that doesn't have it will pass that part of the check. The problem this was fixing was that the "older than" time operator was matching on the "greater than" operator match because this function only looked at the `localName` and not the hint. 

The reverse case could happen if the order of the ops in the `OPERATORS` array was changed for some reason (i.e. a "greater than" operator could hit on the "older than" operator.